Realm/RLMObjectSchema.mm

```diff
@@ -285,11 +285,8 @@ - (BOOL)isEqualToObjectSchema:(RLMObjectSchema *)objectSchema {
     NSArray *otherProperties = objectSchema.properties;
     for (NSUInteger i = 0; i < _properties.count; i++) {
         RLMProperty *p1 = _properties[i], *p2 = otherProperties[i];
-        if (p1.type != p2.type ||
```

Comment:
We don't want to be checking for `indexed` being equal in the place this is used (checking if we can use the same accessor classes).